More operator docs (#2107)
kroxylicious/kroxylicious: docs/modules/configuring/con-configuring-tls-between-client-and-proxy.adoc

[id='con-configuring-virtualkafkacluster-{context}']
= Client-to-proxy TLS

This example shows a `VirtualKafkaCluster`, exposing it to Kafka clients running on the same Kubernetes cluster.
It uses TLS as the transport protocol so that communication between Kafka clients and the proxy is encrypted.

.Example `VirtualKafkaCluster` configuration.
[source,yaml]
----
kind: VirtualKafkaCluster
apiVersion: kroxylicious.io/v1alpha1
metadata:
  name: my-cluster
  namespace: my-proxy
spec:
  proxyRef: # <1>
    name: simple
  targetKafkaServiceRef: # <2>
    name: my-cluster
  ingresses:
    - ingressRef: # <3>
        name: cluster-ip
      tls: # <4>
        certificateRef:
          name: server-certificate
          kind: Secret
----
<1> The `proxyRef` names the `KafkaProxy` resource that this virtual cluster is part of.
  It must be in the same namespace as the `VirtualKafkaCluster`.
<2> The virtual cluster names the `KafkaService` to be proxied.
  It must be in the same namespace as the `VirtualKafkaCluster`.
<3> The virtual cluster can be exposed by one or more ingresses.
  Each ingress names a `KafkaProxyIngress` which must be in the same namespace as the `VirtualKafkaCluster`.

> <4> If the ingress supports TLS, the `tls` property is used to configure the TLS server certificate to be used.

Author: ```suggestion
<4> If the ingress supports TLS, the `tls` property configures the TLS server certificate to use.
```
